src/System.Management.Automation/utils/Verbs.cs

```diff
@@ -1397,6 +1365,62 @@ internal static IEnumerable<VerbInfo> FilterByVerbsAndGroups(string[] verbs, str
             }
         }
 
+        /// <summary>
+        /// Checks if verb type exists in list of groups.
+        /// </summary>
+        /// <param name="groups">The list of groups</param>
+        /// <param name="verbType">The verb type to check.</param>
+        /// <returns>True if verb type was found, False if not found.</returns>
+        private static bool GroupsContainVerbType(string[] groups, Type verbType)
+            => SessionStateUtilities.CollectionContainsValue(
+                groups,
+                GetVerbGroupDisplayName(verbType),
+                StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Enumerates field names from a Verb Type.
+        /// </summary>
+        /// <param name="verbType">The verb type.</param>
+        /// <returns>List of field names.</returns>
+        private static IEnumerable<string> EnumerateFieldNamesFromVerbType(Type verbType)
+        {
+            foreach (FieldInfo field in verbType.GetFields())
+            {
+                if (field.IsLiteral)
+                {
+                    yield return field.Name;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Enumerates field names from all Verb Types.
+        /// </summary>
+        /// <returns>List of field names.</returns>
+        private static IEnumerable<string> EnumerateFieldNamesFromAllVerbTypes()
+        {
+            foreach (Type verbType in VerbTypes)
+            {
+                foreach (string fieldName in EnumerateFieldNamesFromVerbType(verbType))
+                {
+                    yield return fieldName;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Enumerates command verb names.
+        /// </summary>
+        /// <param name="commands">The collection of commands.</param>
+        /// <returns>List of command verb names.</returns>
+        private static IEnumerable<string> EnumerateCommandVerbNames(Collection<CmdletInfo> commands)
+        {
+            foreach (CmdletInfo command in commands)
+            {
+                yield return command.Verb;
```

Comment:
We can remove yield-s in the method too.

Reply:
@iSazonov I think this one is not possible since command verbs are being wrapped in IEnumerable without using LINQ.

Reply:
Ok so simplified this using generic propertly selector so we can process `IEnumerable<T>` and pull the verb out of `CmdletInfo`, so this can be deleted.

Reply:
Oh, it seems last two commits are overkill and unnecessarily complicate the code.

Reply:
@ISazonov Iagree. We can remove the property selector and the other commit. 

If you ask me honestly though I think Linq greatly simplifies it but can just live with yield if there are performance concerns.

Reply:
I've reverted. I think as it is it is fine and simple enough.